src/caikit_nlp_client/grpc_client.py

```diff
@@ -270,3 +279,14 @@ def _make_channel(
             connection,
             grpc.ssl_channel_credentials(**credentials_kwargs),
         )
+
+    @staticmethod
+    def _load_certificate(certificate: Union[None, bytes, str]) -> Optional[bytes]:
+        """If the certificate points to a file, return the contents (plaintext reads).
+        Else return the bytes"""
+        if isinstance(certificate, bytes):
+            return certificate
+        if isinstance(certificate, str):
+            with open(certificate, encoding="utf-8") as secret_file:
+                return bytes(secret_file.read(), "utf-8")
```

Comment:
No need to specify  encoding and then re-converting to bytes:
```suggestion
            with open(certificate, "rb") as fh:
                return fh.read()
 ```

Reply:
Will fix.

Reply:
fixed in [ffe1f0a](https://github.com/opendatahub-io/caikit-nlp-client/pull/79/commits/ffe1f0a9455d0a1848ee79e6e0e8cfa1a8e5a21d)

Reply:
Still loading as utf-8 and then encoding as bytes

Reply:
It should be fixed now.